src/zenml/utils/pipeline_docker_image_builder.py

```diff
@@ -493,6 +571,10 @@ def _generate_zenml_pipeline_dockerfile(
                 f"RUN pip install --default-timeout=60 --no-cache-dir -r {file}"
             )
 
+        for index, packages in extra_index_requirements.items():
```

Comment:
I think in general having the hub plugins in a requirements.txt (maybe some lines of which have an index specifier) would be much nicer as you can then directly look at that file from within the container

Reply:
Unfortunately `--extra-index-url` is unsafe since there is no way to tell pip which index to prefer, i.e., if we use a private repo, someone could register the same package name in PyPi and then `zenml hub install ...` would install that package by default if the version is higher than the one on the hub. It's really stupid actually. See https://github.com/pypa/pip/issues/8606

Reply:
I also don't see how you can specify `--index-url` inside a requirements file from the link you sent, all I see is 
> "The following options have an effect on the entire pip install run, and must be specified on their individual lines. [-i, --index-url](https://pip.pypa.io/en/stable/cli/pip_install/#install-index-url) ..."
which doesn't sound requirement-specific. I guess you could specify the full URL to the wheel but I'm not a big fan of that.

Reply:
I guess we could write all the plugins themselves into another requirements file and then install that via `pip install -r ... --index-url ...`. Would you prefer that option?

Reply:
Mhm you're right, apparently it's a per-file settings, my bad. I'd still prefer two requirements files tbh, and the hub one could even have the `--index-url` inside the file so we can treat them the same from the outside?

Reply:
Sounds good to me, I'll give that a try